File: 53-maximum-subarray/53-maximum-subarray.ts
function maxSubArray(nums: number[]): number {
  const N = nums.length;
  let curr = Number.MIN_SAFE_INTEGER,
      ans = Number.MIN_SAFE_INTEGER;
  
  for(let i = 0; i < N; i++) {
    if(curr + nums[i] > nums[i])
      curr += nums[i];
    
    if(nums[i] > curr)
      curr = nums[i];
    
    ans = Math.max(ans, curr);
  }
  
  return Math.max(ans, curr);
};

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
		int max = 0, ans = Integer.MIN_VALUE;

		for (int i = 0; i < nums.length; i++) {
			max = Math.max(nums[i], max + nums[i]);
			ans = Math.max(ans, max);
		}

		return ans;
	}
}